src/firefly/js/ui/ColorPicker.jsx

```diff
@@ -51,5 +82,7 @@ ColorPickerWrapper.propTypes= {
     callback: PropTypes.func.isRequired,
     color: PropTypes.string.isRequired,
     callbackOnOKOnly: PropTypes.bool.isRequired,
-    callbackOnBoth: PropTypes.bool.isRequired
+    callbackOnBoth: PropTypes.bool.isRequired,
```

Comment:
These 2 flags are really confusing.  Setting callbackOnBoth to `true` will not trigger callback onChanges.  As implemented, only `callbackOnOKOnly` is used.  I am guessing what the original author intended is `callbackOnChange` and `callbackOnOk`.  This way, you can have 4 possible states.

Reply:
I was not thinking about refactoring it. This is the way it has been.